File: 03/java/LetterGrades.java
/* LetterGrades.java is a driver for function letterGrade().
 *
 * CS 214
 * Jonathan Winkle
 * February 2020
 ****************************************************************/
import java.util.Scanner;

public class LetterGrades {

   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your percentage grade (0-100): "); // prompt

      int grade = (keyboard.nextInt());      // read year
      System.out.println(letterGrade(grade)); // display its code
   }

   /***************************************************************
    * yearCode() converts an academic year into its integer code.
    *
    * Receive: year, a string.
    * PRE: year is one of {freshman, sophomore, junior, or senior}.
    * Return: the integer code corresponding to year.
    */

    public static String letterGrade(int grade)
    {
        switch (grade / 10)
        {
            case 10: case 9:
              return "A";
            case 8:
              return "B";
            case 7:
              return "C";
            case 6:
              return "D";
            default:
              return "F";
        }
    }

}

File: 03/clojure/src/letter_grades.clj
;;;; letter_grades.clj is a driver for function letterGrade.
;;;;
;;;; Input: An academic year, one of {freshman, sophomore,
;;;;                                  junior, senior}.
;;;; Output: The code for that academic year
;;;;         (1, 2, 3, 4 for valid entries; 
;;;;          0 for invalid entries).
;;;;
;;;; Usage: clojure -m year_codes
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by:
;;;; Date:
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grades)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; letterGrade() returns the letter grade for a given percentage.
;;; Receive: percentage, an integer.
;;; Precondition: percentage is an integer.
;;; Return: the character letter grade corresponding to the percentage.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;


;; solution using the cond function
(defn letterGrade [percentage]
  (cond
    (> percentage 89)   'A
    (> percentage 79)   'B
    (> percentage 69)   'C
    (> percentage 59)   'D
    :else               'F
  )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our letterGrade() function.
;;; Input: percentage, an integer.
;;; Output: the letter grade corresponding to the percentage.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter your percentage (0-100): ") (flush)
 (let
    [ percentage (Integer/parseInt (read-line)) ]                ; read percentage (integer)
    (println "letter grade: " (letterGrade percentage) "\n")      ; display its code
 )
)

